### DISCUSSION TITLE
Erreur "argument name is invalid" after migration to another server

### DISCUSSION THREAD

**Discussion by Eliophot** [directus/directus#18038] [Archive]
Hello,
I made a dump of the database, uploaded the .env, the extensions and the uploads folder.
I access Directus but when I connect to app (auth) I get this error:
```
[17:36:42.303] ERROR: argument name is invalid
    err: {
      "type": "TypeError",
      "message": "argument name is invalid",
      "stack":
          TypeError: argument name is invalid
```
Do you have an idea?

**Comment by rijkvanzanten** [comment]
Nowhere near enough information to be able to make an educated guess at what the problem is..  What version of Node are you using? What database are you connecting to? What custom extensions are you using?

**Comment by Eliophot** [comment]
Hello,
Sorry, node : v18.15.0 npm 9.5.0
It's MySQL 8.1.17
Extensions have nothing to do with authentication...

**Reply by rijkvanzanten** [reply]
> Extensions have nothing to do with authentication...

The error in your post doesn't have anything to do with auth necessarily either! Does it work if you omit your extensions?

**Comment by Eliophot** [comment]
No I just tried without the extensions, same issue

**Reply by rijkvanzanten** [reply]
What's the full stack trace?

**Reply by Eliophot** [reply]
```
[18:08:22.113] ERROR: argument name is invalid
    err: {
      "type": "TypeError",
      "message": "argument name is invalid",
      "stack":
          TypeError: argument name is invalid
              at Object.serialize (/data/apps/app_eliophot/directus/releases/20230403142457/node_modules/express/node_modules/cookie/index.js:120:11)
              at res.cookie (/data/apps/app_eliophot/directus/releases/20230403142457/node_modules/express/lib/response.js:884:36)
              at /data/apps/app_eliophot/directus/releases/20230403142457/node_modules/directus/dist/auth/drivers/local.js:84:17
    }
```

**Reply by rijkvanzanten** [reply]
What's your value for `REFRESH_TOKEN_COOKIE_NAME`

**Reply by Eliophot** [reply]
ohh
ACCESS_TOKEN_TTL="1d"
REFRESH_TOKEN_TTL="7d"

**Reply by rijkvanzanten** [reply]
Not the values I was asking for 😄 What's your config for `REFRESH_TOKEN_COOKIE_NAME`? 

From what I can tell, the error is thrown by Cookie.js trying to serialize the cookie with an invalid name: 

https://npmdoc.github.io/node-npmdoc-cookie/build/apidoc.html#apidoc.element.cookie.serialize

The name is configured through `REFRESH_TOKEN_COOKIE_NAME`, so I'm assuming that value is somehow invalid

**Reply by Eliophot** [reply]
^^ read to fast
I have no REFRESH_TOKEN_COOKIE_NAME...

**Reply by Eliophot** [reply]
I remove ACCESS_TOKEN_TTL & REFRESH_TOKEN_TTL
And I have a new error : 
```
[18:32:01.456] ERROR: "expiresIn" should be a number of seconds or string representing a timespan
    err: {
      "type": "Error",
      "message": "\"expiresIn\" should be a number of seconds or string representing a timespan",
      "stack":
          Error: "expiresIn" should be a number of seconds or string representing a timespan
              at /data/apps/app_eliophot/directus/releases/20230403142457/node_modules/jsonwebtoken/sign.js:50:15
              at Array.forEach (<anonymous>)
              at validate (/data/apps/app_eliophot/directus/releases/20230403142457/node_modules/jsonwebtoken/sign.js:41:6)
              at validateOptions (/data/apps/app_eliophot/directus/releases/20230403142457/node_modules/jsonwebtoken/sign.js:56:10)
              at module.exports [as sign] (/data/apps/app_eliophot/directus/releases/20230403142457/node_modules/jsonwebtoken/sign.js:165:5)
              at AuthenticationService.login (/data/apps/app_eliophot/directus/releases/20230403142457/node_modules/directus/dist/services/authentication.js:153:52)
              at async /data/apps/app_eliophot/directus/releases/20230403142457/node_modules/directus/dist/auth/drivers/local.js:76:56
    }
```
Well, the problem is obviously with this feature

**Reply by rijkvanzanten** [reply]
All those environment variables have defaults as well, so I'll have to assume you have some other env vars interfering here as well.. I'm afraid I can't really help out with this in much more detail, as this is a self-hosted config problem that's very specific to your exact environment and setup

**Reply by Eliophot** [reply]
Thank you very much for your help, it was indeed a problem of .env, we had left another .env which overloaded the other. Anyway
